torch/csrc/jit/script/module.cpp

```diff
@@ -56,7 +56,7 @@ Value* try_emit_call_to(
              " from a raw graph. File a bug report";
     }
     // TODO: preserve the type information so we don't have to infer it here
-    auto type = incompleteInferTypeFrom(*member);
+    auto type = attemptToRecoverType(*member);
```

Comment:
This is not supposed to be used except for error messages
 
I am going to rename the function so it doesnt get used elsewhere

Reply:
Why would `incompleteInferType` be allowed but `attemptToRecoverType` not be?

Reply:
Neither should be (we should do what's in the TODO) but `attemptToRecoverType` is more correct than `incompleteInferTypeFrom`

Reply:
`incompleteInferType` will error if it encounters an ivalue it doesn't know how to handle, attemptToRecoverType will silently let it through.

Reply:
i dont think we should be letting in things that will silently fail

Reply:
The only time that can happen is with an empty list/dict and that should be throwing an error about an unbound `VarType` somewhere. What's an example of how this would silently fail?

Reply:
An optional list